lib/utils/utils.go

```diff
@@ -452,9 +454,24 @@ func GetFreeTCPPorts(n int, offset ...int) (PortList, error) {
 	return PortList{ports: list}, nil
 }
 
+// HostUUIDExistsLocaly checks if dataDir/host_uuid file exists in local storage.
+func HostUUIDExistsLocaly(dataDir string) bool {
+	_, err := ReadPath(filepath.Join(dataDir, HostUUIDFile))
+	return err == nil
+}
+
 // ReadHostUUID reads host UUID from the file in the data dir
 func ReadHostUUID(dataDir string) (string, error) {
 	out, err := ReadPath(filepath.Join(dataDir, HostUUIDFile))
+	// When running in a Kubernetes Cluster we need to check if the `host_uuid`
+	// was previously written into the Secret.
+	if err != nil && kubernetes.InKubeCluster() {
```

Comment:
My 2c is I would handle "if inside Kubernetes, do this" logic outside of these simple "Read/WriteHostUUID" method that are supposed to just deal with files. I'd let it return an error and handle it on the caller side.

Reply:
fair